### PR TITLE
Board

### PR DESCRIPTION
**Pull request by NitsanLitov**

### CODE REVIEW

File: Board/Board.cs [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+
+using Players;
+using Movement;
+
+namespace ChessBoard
+{
+    class Board
+    {
+        private ChessPiece[,] locationBoard;
+        private List<Type> firstRowPiecesOrder = new List<Type> { typeof(Rook), typeof(Knight), typeof(Bishop), typeof(Queen), typeof(King), typeof(Bishop), typeof(Knight), typeof(Rook) };
```
Comment (EranDaniel98) [inline Board/Board.cs:12]: The order of the pieces to the 2nd or 3rd player is not the same (queen after king), you can reverse the list if you're the 2nd or 3rd player

File: Board/Board.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    this.locationBoard = new ChessPiece[movementBoard.MaxNumber, movementBoard.MaxLetter];
+                }
+                this.CreateChessPieces(movementBoard);
+            }
+        }
+
+        private void CreateChessPieces(MovementBoard movementBoard)
+        {
+            ChessColor color = movementBoard.PlayerColor;
+            chessPiecesByColor[color] = new List<ChessPiece>();
```
Comment (EranDaniel98) [inline Board/Board.cs:65]: I know its working but why not doing:
    List<ChessPiece> chessPieces = new List<ChessPiece>();
    this.chessPiecesByColor[color] =  chessPieces;

Comment (EranDaniel98) [inline Board/Board.cs:65]: you forgot *this* 
this.chessPiecesByColor

File: Board/Board.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+        }
+
+        public void Move((char, int) oldLocation, (char, int) newLocation)
+        {
+            GetPieceByLocation(oldLocation).Move(newLocation);
+        }
+
+        public bool KingWillBeThreatened(ChessPiece piece, (char, int) location)
+        {
+            return false;
```
Comment (EranDaniel98) [inline Board/Board.cs:122]: You didn't write the function logic lol

Comment (NitsanLitov) [reply Board/Board.cs:122]: that will be part of the temp board. code will bw written in a different PR

**Pull request merged**